### ISSUE TITLE
Fix S2259 FN: Auto-Properties don't learn ObjectConstraint

### ISSUE THREAD

**Issue by martin-strecker-sonarsource** [SonarSource/sonar-dotnet#6930]
Instance Fields are tracked by `TrackedSymbol` while auto-properties are not:

The false negative looks like so:
```cs
public object Property { get; set; } // If Property is a field instead, the "Null" constraint is set.

public void M()
{
    Property = null;
    Tag("AfterSetNull", Property); // Auto-property "Property" symbol should have "Null" constraint
    _ = Property.ToString();
    Tag("AfterReadReference", Property);
}
```
Fix:
Add switch expression case
```cs
OperationKindEx.PropertyReference when
    IPropertyReferenceOperationWrapper.FromOperation(operation) is var propertyReference
    && propertyReference.Property.IsAutoProperty()
    && IsStaticOrThis(propertyReference) => propertyReference.Property,
```
to `IOperationExtensions.TrackedSymbol`.

**Comment by antonioaversa** [comment]
It's not clear to me what the automation is doing here:
![image](https://user-images.githubusercontent.com/2676742/230136732-28f08f06-98c8-4d80-a5ef-a80707cf1490.png)

Reassigning @martin-strecker-sonarsource.

**Comment by pavel-mikula-sonarsource** [comment]
You had to do manual change, because there was probably a rebase conflict and action doesn't trigger. GH already has many people complaining about this.

Then Martin requested your review, so it assigned it to you.

Then you approved it and it assigned it back to Tim because he's the author of the PR. It's not Martin-aware.
